crates/subspace-farmer/src/legacy_multi_plots_farm.rs

```diff
@@ -93,61 +91,45 @@ impl LegacyMultiPlotsFarm {
-                            base_directory,
-                            plot_index,
-                            max_plot_pieces,
-                            max_plot_size,
-                            total_pieces,
-                            farming_client,
-                            new_plot,
-                            listen_on,
-                            bootstrap_nodes,
-                            first_listen_on,
-                            enable_farming,
-                            reward_address,
-                            enable_dsn_sync,
-                        })
+        let single_plot_farms = tokio::task::spawn_blocking(move || {
+            plot_sizes
+                .par_iter()
+                .enumerate()
+                .map(|(plot_index, &max_plot_pieces)| {
+                    let base_directory = base_directory.join(format!("plot{plot_index}"));
+                    let farming_client = farming_client.clone();
+                    let new_plot = new_plot.clone();
+                    let listen_on = listen_on.clone();
+                    let bootstrap_nodes = bootstrap_nodes.clone();
+                    let first_listen_on = Arc::clone(&first_listen_on);
+
+                    SinglePlotFarm::new(SinglePlotFarmOptions {
+                        base_directory,
+                        plot_index,
+                        max_plot_pieces,
+                        max_plot_size,
+                        total_pieces,
+                        farming_client,
+                        new_plot,
+                        listen_on,
+                        bootstrap_nodes,
+                        first_listen_on,
+                        enable_farming,
+                        reward_address,
+                        enable_dsn_sync,
                     })
-                    .collect::<anyhow::Result<Vec<_>>>()?;
-
-                for (farm, node_runner) in single_plot_farm_instantiations {
-                    if node.is_none() {
-                        node = Some(farm.node.clone());
-                    }
-                    networking_node_runners.push(node_runner);
-                    single_plot_farms.push(farm);
-                }
-
-                Ok::<_, anyhow::Error>((single_plot_farms, networking_node_runners, node))
-            })
-            .await
-            .expect("Not supposed to panic, crash if it does")?;
+                })
+                .collect::<anyhow::Result<Vec<_>>>()
+        })
+        .await
+        .expect("Not supposed to panic, crash if it does")?;
 
         // Start archiving task
         let archiving = Archiving::start(
             farmer_metadata,
             object_mappings,
             archiving_client,
-            enable_dsn_archiving
-                .then(|| node.expect("Always set, as we have at least one networking instance")),
+            enable_dsn_archiving.then(|| single_plot_farms[0].node.clone()),
```

Comment:
I think we should either don't use indexing or allow some simple unwraps. :upside_down_face:

Reply:
I agree, not using indexing would be better, fixed